vulnerabilities/insufficient-access-control.md

```diff
@@ -1,3 +1,159 @@
 ## Insufficient Access Control
 
-Access control is often imperative in management and ownership of smart contracts. It's important to consider ways in which access control may be circumvented, or insufficiently implemented and the corresponding consequences.
\ No newline at end of file
+Access control is often imperative in management and ownership of smart contracts. It's important to consider ways in which access control may be circumvented, or insufficiently implemented and the corresponding consequences. Contracts may allow any user to execute sensitive functions without verifying their authorization status. For example, a contract may have functions that transfer ownership, mint tokens, or modify critical state variables without checking if the caller has the appropriate permissions. 
+
+Improper implementation of access control can lead to severe vulnerabilities, allowing unauthorized users to manipulate the contract’s state or even drain its funds. Some of them are listed below:
+
+#### Unrestricted Initialization Function   
+```solidity
+function initContract() public {
+    owner = msg.sender;
+}
+```
+The ``initContract()`` function sets the caller as the owner but lacks any checks to prevent re-initialization. An attacker can call this function to take ownership of the contract, thereby gaining control over its funds and functionalities.
+
+#### Overpowered Roles
+```solidity
+// Using OpenZeppelin's Ownable library
+function criticalFunction() public onlyOwner {
+    // Critical logic here
+}
+```
+If the contract assigns multiple roles with ``onlyOwner`` privileges, it increases the attack surface. An attacker compromising a single owner account could execute critical functions.
```

Comment:
This shouldn't be possible with OZ Ownable though

Reply:
My bad. OZ Ownable is the fix for role based access control with multiple onlyOwner privileges. Will fix.